marlowe-playground-client/src/Simulation/BottomPanel.purs

```diff
@@ -0,0 +1,495 @@
+                  (Map.toUnfoldable tokenMap)
+              )
+          )
+          (Map.toUnfoldable mon)
+    in
+      concatMap asTuple payments
+
+  bindingsData =
+    let
+      (bindings :: Array _) = state ^. (_marloweState <<< _Head <<< _state <<< _boundValues <<< to Map.toUnfoldable)
+
+      asTuple (Tuple (ValueId valueId) value) = show valueId /\ show value /\ mempty /\ mempty /\ mempty /\ unit
+    in
+      map asTuple bindings
+
+  tableRow title emptyMessage a b c d e [] = emptyRow title emptyMessage
+
+  tableRow title emptyMessage a b c d e rowData = headerRow title a b c d e <> foldMap (\dataTuple -> uncurry5 row dataTuple) rowData
+
+  headerRow title a b c d e =
+    [ div [ classes [ rTableCell, first, Classes.header ] ] [ text title ] ]
+      <> map (\x -> div [ classes [ rTableCell, rTableCell, Classes.header ] ] [ text a ]) [ a, b, c, d, e ]
+
+  row a b c d e =
+    [ div [ classes [ rTableCell, first ] ] [] ]
+      <> map (\x -> div [ class_ rTableCell ] [ text x ]) [ a, b, c, d, e ]
+
+  emptyRow title message =
+    [ div [ classes [ rTableCell, first, Classes.header ] ]
+        [ text title ]
+    , div [ classes [ rTableCell, rTableEmptyRow, Classes.header ] ] [ text message ]
+    ]
+
+  displayWarning' (TransactionNonPositiveDeposit party (AccountId accNum owner) tok amount) =
+    [ div [ classes [ rTableCell, first ] ] []
+    , div [ class_ (ClassName "RTable-2-cells") ] [ text "TransactionNonPositiveDeposit" ]
+    , div [ class_ (ClassName "RTable-4-cells") ]
+        [ text $ "Party " <> show party <> " is asked to deposit " <> show amount
+            <> " units of "
+            <> show tok
+            <> " into account "
+            <> show accNum
+            <> " of "
+            <> show owner
+            <> "."
+        ]
+    ]
+
+  displayWarning' (TransactionNonPositivePay (AccountId accNum owner) payee tok amount) =
+    [ div [ classes [ rTableCell, first ] ] []
+    , div [ class_ (ClassName "RTable-2-cells") ] [ text "TransactionNonPositivePay" ]
+    , div [ class_ (ClassName "RTable-4-cells") ]
+        [ text $ "The contract is supposed to make a payment of "
+            <> show amount
+            <> " units of "
+            <> show tok
+            <> " from account "
+            <> show accNum
+            <> " of "
+            <> show owner
+            <> " to "
+            <> ( case payee of
+                  (Account (AccountId accNum2 owner2)) -> "account " <> show accNum2 <> " of " <> show owner2
+                  (Party dest) -> "party " <> show dest
+              )
+            <> "."
+        ]
+    ]
+
+  displayWarning' (TransactionPartialPay (AccountId accNum owner) payee tok amount expected) =
+    [ div [ classes [ rTableCell, first ] ] []
+    , div [ class_ (ClassName "RTable-2-cells") ] [ text "TransactionPartialPay" ]
+    , div [ class_ (ClassName "RTable-4-cells") ]
+        [ text $ "The contract is supposed to make a payment of "
```

Comment:
Time for `mconcat`? 🤔

Reply:
what is the advantage of `mconcat` in this case?

Reply:
`w <> x <> y <> z`

vs.

`mconcat [ w, x, y, z ]`

For really long lists the latter feels neater to me, but it's a preference thing. Can't justify it. Up to you.

Reply:
yeah, I just don't see how replacing `<>` with `,` is better.